contrib/initramfs/zfsunlock

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+set -eu
+echo "Waiting for zfs init script to complete, press Ctrl-C to exit into shell ..."
+while [ ! -e /run/zfs_fs_name ] && [ ! -e /run/zfs_unlock_complete ]; do
+	sleep 1;
+done
+if [ -e /run/zfs_unlock_complete ]; then
+	exit 0
+fi
+echo
+echo "Unlocking encrypted ZFS pools ..."
+echo "Enter the password or press Ctrl-C to exit into shell."
+echo
+zfs_fs_name=""
+fs_password_accepted=false
+while true; do
+	while [ "$fs_password_accepted" = "true" ] && [ "$(cat /run/zfs_fs_name)" = "$zfs_fs_name" ] && [ ! -e /run/zfs_unlock_complete ]; do
```

Comment:
Can you elaborate a bit on what's going on here or confirm if my understanding is correct? Is this saying: if the user's password was accepted, sleep until either the unlocking is done or until `/run/zfs_fs_name` changes (i.e. we have another filesystem to unlock)?

If so, this would probably be cleaner to put in the code path where the password is accepted. That would eliminate the need to check `$fs_password_accepted` in this loop. Plus, since that's the only use for that variable, it could be removed entirely. So it would be something like.

```
...
echo "Please reconnect in a while."
... the kill thing ...
# Wait for another filesystem to unlock.
while [ "$(cat /run/zfs_fs_name)" = "$zfs_fs_name" ]; do
    if [ -e /run/zfs_unlock_complete ]; then
        break # or exit 0, your choice
    fi
    sleep 1
done
```

Then, the first check in the main loop is for `/run/zfs_unlock_complete` so your `while true` can become `while [ ! -e /run/zfs_unlock_complete]`.

Reply:
Maybe I forgot to mark this unresolved.

Reply:
I've already moved the messages output and break to anotehr place.